src/Cache.cpp

```diff
@@ -175,6 +177,9 @@ Cache::setup()
                             .arg(QStandardPaths::writableLocation(QStandardPaths::CacheLocation))
                             .arg(QString::fromUtf8(localUserId_.toUtf8().toHex()));
 
+        UserSettings settings;
+        statePath += settings.profile();
```

Comment:
Please make statePath and cacheDirectory_ the same. Currently this makes this delete the wrong directory on Logout. Even better, if you just remove the statePath.